proxy/util/cloudsqlutil.go

```diff
@@ -43,3 +46,19 @@ func SplitName(instance string) (project, region, name string) {
 		return spl[0], spl[1], spl[2]
 	}
 }
+
+// Validate verifies that instances are in the expected format and include
+// the appropriate components.
+func Validate(instance string) (bool, error) {
```

Comment:
Does this function need to live here or can it live in `client.go`, where it's used?

Also, there should be some unit tests around it.

Reply:
Did you mean to use this function here as well?

https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/02b93080ece77f30fe26212998c0f5f1c93e1ea1/cmd/cloud_sql_proxy/proxy.go#L228-L238

Reply:
I could put the function in `client.go`, then also call it from `fuse.go`. From the discussion [below](https://github.com/GoogleCloudPlatform/cloudsql-proxy/pull/871#discussion_r684474107), this code should take place in its own PR. (I'm removing it from here.)

Reply:
Sonuds good.

Reply:
Let's get that PR out first and then rebase this on top.